great_expectations/core/expectation_suite.py

```diff
@@ -937,6 +939,33 @@ def _get_expectations_by_domain_using_accessor_method(
 
         return expectation_configurations_by_domain
 
+    def render(self) -> None:
+        """
+        Renders content using the atomic prescriptive renderer for each expectation configuration associated with
+           this ExpectationSuite to ExpectationConfiguration.rendered_content.
+        """
+        for expectation_configuration in self.expectations:
+            inline_renderer_config: Dict[str, Union[str, ExpectationConfiguration]] = {
+                "class_name": "InlineRenderer",
+                "render_object": expectation_configuration,
+            }
+            module_name = "great_expectations.render.renderer.inline_renderer"
+            inline_renderer = instantiate_class_from_config(
```

Comment:
Typically I would avoid this method, as we prefer to use typed objects, but this solves circular import issues without decoupling a lot of packages.

Reply:
Is there value in using a `TypedDict`?

Reply:
https://mypy.readthedocs.io/en/stable/runtime_troubles.html?highlight=if%20type_checking#typing-type-checking

Is this something that can be resolved with a `if TYPE_CHECKING:` conditional?
Happy to sync on this.

Reply:
The `TypedDict` would need to be imported from somewhere (`inline_renderer.py`) which will cause circular imports.

The `TYPE_CHECKING` solution only works for annotations right? The problems seem deeper than that.

Reply:
@NathanFarmer Yes `if TYPE_CHEKCING:` is for annotations. Which the `TypeDict` can be. You don't need to actually instantiate the TypeDict class.
```python
if TYPE_CHECKING:
  from inline_render import MyTypeDict


...

my_type_dict: 'MyTypeDict' = {"foo": "bar"}

```

Reply:
Sorry for this additional ask but could we go ahead and remove `core/expectation_suite.py` from the `mypy` exclude list?
https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L45
According to this it only has `10` outstanding typing errors.
I'm also happy to submit a small PR against your branch with these fixes.

Reply:
This is taken care of. Thanks @Kilo59!